src/segmantic/seg/dataset.py

```diff
@@ -134,26 +180,73 @@ def load_from_json(
             "training": [{"image": "image/*.nii.gz", "label": "label/*.nii.gz"}],
         }
         """
+
         if isinstance(file_path, (Path, str)):
             file_path = [file_path]
 
-        data_dicts: List[Dict[str, Path]] = []
+        data_dicts_train: List[Dict[str, Path]] = []
+        data_dicts_val: List[Dict[str, Path]] = []
+        data_dicts_test: List[Dict[str, Path]] = []
 
-        for p in (Path(f) for f in file_path):
-            training = json.loads(p.read_text())["training"]
-            for d in training:
+        for p in [Path(f) for f in file_path]:
+            ds = json.loads(p.read_text())
+            training = ds["training"]
+            validation = ds["validation"]
+            test = ds["test"] if "test" in ds else []
+
+            for t in training:
```

Comment:
this is a lot of code duplication (essentially the same for `training`, `validation`, `test`